js/ui.js

```diff
@@ -1,10 +1,56 @@
-export default {
-	display: display,
-	showError: showError,
-	reflectParameterChange: reflectParameterChange,
-	displayCommunesFetchResults: displayCommunesFetchResults,
+import transForms from './transForms.js'
+
+const getForm = () => document.querySelector('.SGMAPembauche form')
+
+const getOutputVariables = () => getForm().action
+
+function collectInput(form) {
+	let input = collectFormFields(form)
+
+	Object.assign(input, addBooleanParameters())
+
+	// Additional parameters
+	var today = new Date()
+	input['contrat_de_travail_debut'] = today.getFullYear() + '-' + (today.getMonth() + 1)
+
+	return input
 }
 
+// Collect form fields that will be the base for the API input
+const collectFormFields = (form) =>
+		[ ...form.elements ].reduce(
+			(memo, element) => {
+				if (! element.name)
+					return memo
+
+				/* 	Some of the form values must be transformed
+						before being sent through the request */
+				const value = transForms.reduce(
+					(value, transformer) => {
+						const [ test, transformed ] = transformer(element, value)
+						return test ? transformed : value
```

Comment:
Why not simply let it to the transform function to return the modified version or be a pass-through? KISS